addons/arsenal/defines.hpp

```diff
@@ -460,3 +460,6 @@ if (!isNil QGVAR(customRightPanelButtons)) then {\
     _contentPanelCtrl lnbSetPicture [[_newRow, 7], getText (configFile >> "cfgVehicles" >> ((_loadout select 5) select 0) >> "picture")];\
     _contentPanelCtrl lnbSetPicture [[_newRow, 8], getText (configFile >> "cfgWeapons" >> (_loadout select 6) >> "picture")];\
     _contentPanelCtrl lnbSetPicture [[_newRow, 9], getText (configFile >> "cfgGlasses" >> (_loadout select 7) >> "picture")];
+
+#define ALL_BOX_VIRTUAL_ITEMS\
+    (flatten (GVAR(currentBox) getVariable [QGVAR(virtualItems),[]]))
```

Comment:
How expensive is `flatten`? `_fnc_fill_right_Container` is called for every list item that is added which then uses `ALL_BOX_VIRTUAL_ITEMS` every time. Might be better to just `flatten` the virtual items once.

Reply:
Good point.